sympy/logic/boolalg.py

```diff
@@ -679,20 +679,21 @@ class And(LatticeOp, BooleanFunction):
 
     @classmethod
     def _new_args_filter(cls, args):
-        newargs = []
-        rel = []
         args = BooleanFunction.binary_check_and_simplify(*args)
-        for x in reversed(args):
+        args = LatticeOp._new_args_filter(args, And)
+        newargs = []
+        rel = set()
+        for x in args:
```

Comment:
```suggestion
        for x in ordered(arg)s:
```

Since we are going to be filtering args, let's make it canonical:
```python
>>> set([And(*i) for i in permutations((l,g,le,ge))])
{(A >= B) & (A > B)}
```
(`ordered` is already imported so this diff can be added without additional edits; the test of permutations should be added, though)

Reply:
I'm a bit confused by this bit of code. Previously there was `reversed` rather than `ordered`. Removing reversed caused the only test failure that I saw here. Do you remember why the `reversed` was added?

Reply:
Unfortunately there was a syntaxerror with this change.

Reply:
I think the `reversed` may have been an attempt to make something matching a canonical form be the first form that was seen. But I think canonically predictable is better than predictable.